### PR TITLE
Make source paths easier to configure

### PR DESCRIPTION
**Pull request by Stricted**
Change-Id: I44ab9df74bc4042bcdac743e4f1e5e572a3d900b

### CODE REVIEW

File: lineage-14.1.groovy [hunk 1]
```diff
@@ -131,7 +130,7 @@ timestamps {
         withCredentials([string(credentialsId: '3ad6afb4-1f2a-45e9-94c7-b2b511f81d50', variable: 'UPDATER_API_KEY')]) {
```
Comment (Stricted) [inline lineage-14.1.groovy]: im not sure if we can pass the value for credentialsId as build parameter too
would be great if we can (makes the script usable for more ppl without modifying it too much)

**Comment by harryyoud** [comment]
I'm slightly against adding more parameters to the lineage-14.1, because I have multiple users on Jenkins and there is no need for them to be able to specify the mirror, ccache, or build tree parameters.

I like the idea of using variables throughout though. Perhaps the definition of them could be moved to the top of the lineage-14.1 file? They would have to be defined as Groovy variables in order to persist through the various shell environments launched later

**Comment by Stricted** [comment]
i was thinking of this too, also i was thinking about an "settings.groovy" which gets included/loaded
which would give you the advantage of using the build script as it is and only modify the local settings file (like on midnightsnack)

this here was my first thinking about that (more ore less as an discussion base)

**Comment by harryyoud** [comment]
To be honest, I think just moving the variables to the top of the script is the best way. It's unlikely there'll be merge conflicts in this area anyway

File: lineage-14.1.groovy (hunk 1 above)
Comment (harryyoud) [inline lineage-14.1.groovy]: I'd rather not pass these as parameters if possible. Again, this is going to be another place where merge conflicts are unlikely to happen

**Comment by harryyoud** [comment]
@Stricted happy with this?

**Comment by Stricted** [comment]
sure, better as editing tons of places for the path

**Comment by Stricted** [comment]
dosent seem to work, i created an test job on my jenkins with the following pipeline
```
def TEST = "lalala"
node("master"){
	timestamps {
		stage('test') {
			sh '''#!/bin/bash +x
				echo "foo - $TEST"
				echo "bar - ${TEST}"
			'''
		}
	}
}
```

it should print
```
foo - lalala
bar - lalala
```
but it prints
```
05:39:58 [test] Running shell script
05:39:58 foo - 
05:39:58 bar - 
```

so it dosent work

File: lineage-14.1.groovy [hunk 2]
```diff
@@ -1,5 +1,9 @@
 String calcDate() { ['date', '+%Y%m%d'].execute().text.trim()}
 
+def MIRRORTREE = "/mnt/Media/Lineage"
```
Comment (Stricted) [inline lineage-14.1.groovy]: shouldnt the name here match with the ones used below?

Comment (harryyoud) [inline lineage-14.1.groovy]: Yes, my mistake

**Comment by harryyoud** [comment]
Use double quotes instead of single quotes for your sh step:

```
sh """
    echo $TEST
"""
```

**Comment by harryyoud** [comment]
Okay. Fixed those. Look good now?

**Comment by harryyoud** [comment]
Just tested this here, and it worked fine.

**Comment by Stricted** [comment]
looks good for me

**Pull request merged**